aws-auth-cognito/src/main/java/com/amplifyframework/auth/cognito/CredentialStoreClient.kt

```diff
@@ -72,11 +72,12 @@ internal class CredentialStoreClient(configuration: AuthConfiguration, context:
                     is CredentialStoreState.Idle -> {
                         val success = capturedSuccess
                         val error = capturedError
-                        if (success != null) {
-                            credentialStoreStateMachine.cancel(token)
+                        if (success != null && token != null) {
+                            credentialStoreStateMachine.cancel(token as StateChangeListenerToken)
+                            token = null
                             onSuccess(success)
-                        } else if (error != null) {
-                            credentialStoreStateMachine.cancel(token)
+                        } else if (error != null && token != null) {
```

Comment:
Token also has to be set to null for the error case.

Reply:
Good catch. Done.